### PR TITLE
Product Page: refactor grid layout and update styles

### PR DESCRIPTION
**Pull request by xelaint**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
```

## What is the current behavior?
Grid layout on product page is too complicated with grids nested within grids. Some containers do not require the use of css grid.

Issue Number: N/A

## What is the new behavior?
Simplify grid layout and remove `display: grid;` on containers that can be styled simpler.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: apps/foundation-demo/src/app/product/components/product/product.component.scss [hunk 1]
```diff
@@ -63,54 +64,41 @@
...
     font-size: $medium-font-size;
   }
 
   &__qty {
-    display: grid;
-    margin-left: 15px;
-    grid-gap: 25px;    
-    grid-template-columns: auto 1fr;
+    display: flex;
+    flex-wrap: wrap;
```
Comment (lderrickable) [inline apps/foundation-demo/src/app/product/components/product/product.component.scss]: not needed, since qty will never overflow or need to wrap

**Comment by lderrickable** [comment]
I really like that you added the `product__details` class. Makes it a lot cleaner. I think it would be nice if product-details was its own component though, because there's a lot going on in `product` right now. I'll leave it up to you though, whether you want to do it in this PR or I could do it in another.

**Comment by xelaint** [comment]
> I really like that you added the `product__details` class. Makes it a lot cleaner. I think it would be nice if product-details was its own component though, because there's a lot going on in `product` right now. I'll leave it up to you though, whether you want to do it in this PR or I could do it in another.

Let's leave this for another PR?

File: apps/foundation-demo/src/app/product/components/product/product.component.scss [hunk 2]
```diff
@@ -3,55 +3,56 @@
 @mixin product-page-grid() {
   display: grid;
   grid-gap: 25px;
+  grid-template-columns: 1fr;
+  padding: 25px 0 50px 0;
 
-  @include breakpoint(medium) {
-    padding: 25px;
+  &__image-gallery {
+    order: 2;
```
[the remark this answers is not in the document]
Comment (lderrickable) [reply apps/foundation-demo/src/app/product/components/product/product.component.scss]: ohh, you're right. I was only looking at it in a larger view when the `order` got overridden by `grid-template-areas`

**Pull request merged**